src/stream_base.h

```diff
@@ -275,10 +278,11 @@ class StreamResource {
 
 class StreamBase : public StreamResource {
  public:
-  // 0 is reserved for the BaseObject pointer.
-  static constexpr int kStreamBaseField = 1;
-  static constexpr int kOnReadFunctionField = 2;
-  static constexpr int kStreamBaseFieldCount = 3;
+  enum InternalFields {
+    kStreamBaseField = BaseObject::kInternalFieldCount,
```

Comment:
This doesn't look right either for much the same reason, but also because it requires you to check that `StreamResource` isn't a `BaseObject` or uses fields itself.

Aside: I don't really understand why `StreamResource` is a separate class. It looks like it can be merged into `StreamBase`.

Reply:
`StreamResource` and `StreamBase` are similar to `StreamReq` in that the current uses will also derive from `BaseObject`. I'm updating the enums to have their own `kSlot` definitions with a comment that those mirror `BaseObject::InternalFields::kSlot` and why. These parts could be revisited later in a separate PR.